src/components/Steps/Steps.tsx

```diff
@@ -1,30 +1,50 @@
-import React, { useRef, forwardRef } from "react";
-import PropTypes from "prop-types";
+import React, { FC, forwardRef, ReactElement, useRef } from "react";
 import cx from "classnames";
 import { BEMClass } from "../../helpers/bem-helper";
 import { NOOP } from "../../utils/function-utils";
 import useMergeRefs from "../../hooks/useMergeRefs";
 import { StepsHeader } from "./StepsHeader";
-import { STEPS_GALLERY_TYPE, STEPS_CSS_BASE_CLASS, STEPS_TYPES } from "./StepsConstants";
+import { STEPS_CSS_BASE_CLASS, StepsType } from "./StepsConstants";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import { ButtonProps } from "../Button/Button";
 import "./Steps.scss";
 
 const bemHelper = BEMClass(STEPS_CSS_BASE_CLASS);
 
-const Steps = forwardRef(
+interface StepsProps extends VibeComponentProps {
+  /**
+   * The index of the current displayed step
+   */
+  activeStepIndex?: number;
+  /**
+   * A callback which called when the active step is changed
+   */
+  onChangeActiveStep?: (e: React.MouseEvent, stepIndex: number) => void;
+  areNavigationButtonsHidden?: boolean;
+  steps?: ReactElement[];
+  type?: StepsType;
+  isOnPrimary?: boolean;
+  isContentOnTop?: boolean;
+  areButtonsIconsHidden?: boolean;
+  backButtonProps?: ButtonProps;
+  nextButtonProps?: ButtonProps;
+}
+
+const Steps: FC<StepsProps> & { types?: typeof StepsType } = forwardRef(
```

Comment:
As far as I remember this is not the right type for a component written using using forwardRef, and it should not compile. Is it compile? if not I think using VibeComponent type will fix it. We usually use it for this case.

Reply:
Fixed this one